prbt_hardware_support/spec.dox

```diff
@@ -0,0 +1,110 @@
+ *  @brief System Specification
+ *
+ *  @subsection ISO-10218-1
+ *
+ *
+ *
+ * @subsubsection sto_stop1 Safe torque off (STO) and  Stop 1
+ *
+ * @definereq{Stop1_Trigger}
+ *  - A Stop 1 is triggered, if at least one of the following cases is true:
+ *    - the state of the STO changes from TRUE -> FALSE,
+ *    - connection to safety controller is lost,
+ *    - the messages informing about the STO (coming from the
+ *      safety controller) is incomplete or in any way corrupted,
+ *    - the communication protocol used by the safety controller
+ *      does not fulfill the expected specification.
+ *
+ * @definereq{No_new_commands_during_STO_false}
+ *  - As long as the STO is FALSE, no new motions commands are allowed.
+ *
+ * @definereq{Hold_driver_if_STO_false}
+ *  - If the STO changes from TRUE -> FALSE the driver is halt.
+ *
+ * @definereq{Recover_driver_after_STO_false}
+ *  - If the STO changes from FALSE -> TRUE, the driver is recovered.
+ *
+ *
+ * @definereq{No_Startup_if_controller_hold_missing}
+ *  - The system is not allowed to complete the startup procedure until
+ *    the controller hold service is available.
+ *
+ * @definereq{No_Startup_if_driver_halt_missing}
+ *  - The system is not allowed to complete the startup procedure until
+ *    the driver halt service is available.
+ *
+ * @definereq{Controller_service_unhold_optional}
+ *  - The system can still be started even if the controller unhold service
+ *    is missing.
+ *
+ * @definereq{Driver_service_recover_optional}
+ *  - The system can still be started even if the driver recover service
+ *    is missing.
+ *
+ *
+ *
+ * @subsubsection brake_test Braketest
+ *
+ * @definereq{Is_BrakeTest_required_mechanism}
+ *  - The system provides a mechanism which allows to determine if
+ *    a braketest is requested by the safety controller.
+ *
+ *  - The mechanism continues to return the last received value,
+ *    if at least one of the following conditions is true:
+ *    - the connection to the safety controller is lost,
+ *    - the messages informing about the operation mode (coming from the
+ *      safety controller) is incomplete or in any way corrupted,
+ *    - the communication protocol used by the safety controller
+ *      does not fulfill the expected specification.
+ *
+ * @definereq{Execute_BrakeTest_mechanism}
+ * - The system provides a mechanism which allows to execute a brake test.
+ * - The brake test is performed for all axes.
+ * - The brake test mechanism returns a status stating if the
+ *    brake test was successful xor what error occured.
+ * - At the beginning of the brake test it is checked if the robot is
+ *    currently moving. In case the robot is moving, the brake test is not
+ *    performed and an error is retured.
+ * - As long as the brake test is performed no motion is allowed. Commands
+ *    send during the execution of the brake test are rejected.
+ * - Brake tests are only performed for drives which define that they require
```

Comment:
I know you want to keep this high level but maybe a hint to the config file?

Reply:
Done.